api/php/DBgateways/RestaurantDbGateway.php

```diff
@@ -0,0 +1,73 @@
+<?php
+
+include_once "php/config.php";
+
+class RestaurantDbGateway {
+
+    // private $servername = "localhost";
+    // private $username = "root";
+    // private $password = "foody";
+    // private $db = "foody";
+
+
+    private $dbConnection = null;
+
+    public function __construct() {
+        $this->dbConnection = new DbConnecter();
+    }
+
+    // public function getRestaurants () {
+
+    // }
+
+    public function validateRestaurant (string $restaurantName, string $restaurantAddress, string $restaurantPhone, string $restaurantWebsite) {
+        $restaurantName = $this->dbConnection->cleanInput($restaurantName);
+        $restaurantAddress = $this->dbConnection->cleanInput($restaurantAddress);
+        $restaurantPhone = $this->dbConnection->cleanInput($restaurantPhone);
+        $restaurantWebsite = $this->dbConnection->cleanInput($restaurantWebsite);
+        $restaurantName = $this->dbConnection->cleanInput($restaurantName);
+        $restaurantAddress = $this->dbConnection->cleanInput($restaurantAddress);
+
+        $sql = "CALL foody.sp_validateRestaurant ('$restaurantName', '$restaurantAddress', '$restaurantPhone', '$restaurantWebsite');";
+
+        $result = $this->dbConnection->returnQuery($sql);
+
+        $output = (int)mysqli_fetch_assoc($result)["restaurantId"];
+
+
+        $this->dbConnection->cleanSPResults();
+
+        return $output;
+    }
+
+    public function saveRestaurantReview (int $restaurantId, int $userId, string $review, int $deliciousnessScore, int $serviceScore, int $experienceScore, int $pricingScore, float $pricingValue) {
+        $review = $this->dbConnection->cleanInput($review);
+
+        $sql = "CALL foody.sp_saveReview ($restaurantId, $userId, '$review', $deliciousnessScore, $serviceScore, $experienceScore, $pricingScore, $pricingValue);";
+
+        $result = $this->dbConnection->returnQuery($sql);
+        $output = mysqli_fetch_assoc($result)["result"];
+
+        $this->dbConnection->cleanSPResults();
+
+        return $output;
```

Comment:
I think this method name can be rewritten. "saveRestaurantReview" is inferring that we are just inserting into the Database table. However, this method is returning a value. This name is a little confusing, what are we returning? It is hard to tell because the query is not written here.